### ISSUE TITLE
Add support for font-agnostic pixel-perfect rendering of ASCII box drawing glyphs.

### ISSUE THREAD

**Issue by colepeters** [warpdotdev/Warp#1940]
### Describe the bug

Text rendering in Warp appears to be messing with ASCII box drawing. For example, here's some Neovim UI which uses box drawing characters in iTerm (see the borders on completion and signature popovers, and the box drawing characters used in the status line and file explorer):
<img width="1011" alt="image" src="https://user-images.githubusercontent.com/1713932/194616787-588c8123-eafc-46f6-80e3-f78b0dcea011.png">

Here's the same UI in Warp, with line height set to the default 1.2:
<img width="1010" alt="image" src="https://user-images.githubusercontent.com/1713932/194617131-e1c585b6-b89d-4231-b031-dba381f4d786.png">

Even with the line height set to 1 (which makes for really cramped reading), gaps still exist, and it also creates misalignment in the status line:
<img width="1010" alt="image" src="https://user-images.githubusercontent.com/1713932/194616835-636105be-b3fe-438d-9072-06d59fad0781.png">

Aside from the noticeable vertical gaps, there's also some strange horizontal collisions, like this:
<img width="300" alt="image" src="https://user-images.githubusercontent.com/1713932/194617351-f40dfdf2-52e0-40cd-9b8e-5ed40248066b.png">

Hope this helps, please let me know if I can provide further info!

### To Reproduce

Initialize any UI which uses ASCII box drawing characters. (For a basic example, try [vtop](https://www.npmjs.com/package/vtop).)

### Expected behaviour

Box drawing characters should draw connected lines.

### Screenshots

See above

### Operating System

MacOS

### OS Version

11.6

### Shell Version

zsh 5.7.1

### Warp Version

v0.2022.10.04.08.05.stable_00

### Does this block you from using Warp daily?

No

**Comment by colepeters** [comment]
Following up on 2 questions from @vorporeal in #1294:

1. The font in these screenshots is IBM Plex Mono
2. If I disable iTerm’s custom box drawing logic, I do get similar results to Warp (see below) — so I'm not sure if this means that what I'm seeing in Warp is not so much a bug but a lack of a feature 😅
    <img width="948" alt="image" src="https://user-images.githubusercontent.com/1713932/194968006-04da134a-6c8a-4661-9391-0b87dc5c3f94.png">

**Comment by unphased** [comment]
Hey @colepeters Just curious what is the tool you're using to display the file tree in your screenshots? Is it a nvim plugin or feature of some sort? Thanks.

**Comment by dannyneira** [comment]
Thanks for submitting, we'll post any updates here.

**Comment by Zabriskije** [comment]
+1
Box-drawing characters have rendering issues in a simple `tree` command too (left iTerm, right Warp):

![enhancetree](https://github.com/warpdotdev/Warp/assets/101254295/32f17653-7d13-4535-8e69-f4754cf1d884)

Both using [Liga SFMono Nerd Font](https://github.com/shaunsingh/SFMono-Nerd-Font-Ligaturized).

**Comment by vorporeal** [comment]
Yeah, this is because iTerm ignores the actual box drawing glyphs in the font you choose and renders the lines itself.  We don't do this yet (but plan to in the future!), so the "niceness" of the rendering at the moment depends on the font you're using.

I updated the labels and title for this issue to better reflect that this is a not-yet-implemented feature and not a bug, per-se.

**Comment by marlonjames71** [comment]
Any updates on this?
